src/H5L.c

```diff
@@ -1614,11 +1432,10 @@ H5Lis_registered(H5L_type_t id)
         HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid link type id number")
 
     /* Is the link class already registered? */
-    for (i = 0; i < H5L_table_used_g; i++)
-        if (H5L_table_g[i].id == id) {
-            ret_value = TRUE;
-            break;
-        } /* end if */
+    if (H5L_is_registered(id, &is_registered) < 0)
+        HGOTO_ERROR(H5E_LINK, H5E_BADTYPE, FAIL, "could not determine registration status of UD link type")
+
+    ret_value = is_registered ? TRUE : FALSE;
```

Comment:
From the Dept. of Redundant Redundancy. ;-) ;-)     [Could just assign the boolean variable, instead of checking if it was true]

Reply:
I tend to do this solely because htri_t and hbool_t are distinct types. In practice, that shouldn't (?) matter, but I try not to assign different types to each other.

Reply:
> I tend to do this solely because htri_t and hbool_t are distinct types. In practice, that shouldn't (?) matter, but I try not to assign different types to each other.

That doesn't really apply here, since this code is still assigning a boolean to an htri_t. :-)